packages/task/src/browser/task-service.ts

```diff
@@ -201,16 +203,39 @@ export class TaskService implements TaskConfigurationClient {
             this.messageService.info(`Task '${taskIdentifier}' has been started.`);
         });
 
-        this.taskWatcher.onOutputProcessed((event: TaskOutputProcessedEvent) => {
+        this.taskWatcher.onOutputProcessed(async (event: TaskOutputProcessedEvent) => {
             if (!this.isEventForThisClient(event.ctx)) {
                 return;
             }
             if (event.problems) {
+                const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+                // check if the task is active
+                const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+                    const taskConfig = taskInfo.config;
+                    return this.taskDefinitionRegistry.compareTasks(taskConfig, event.config);
+                });
+                const isTaskActiveAndOutputSilent = matchedRunningTaskInfo &&
+                    matchedRunningTaskInfo.config.presentation && matchedRunningTaskInfo.config.presentation.reveal === RevealKind.Silent;
                 event.problems.forEach(problem => {
                     const existingMarkers = this.problemManager.findMarkers({ owner: problem.description.owner });
                     const uris = new Set<string>();
                     existingMarkers.forEach(marker => uris.add(marker.uri));
                     if (ProblemMatchData.is(problem) && problem.resource) {
+                        // When task.presentation.reveal === RevealKind.Silent, put focus on the terminal only if it is an error
+                        if (isTaskActiveAndOutputSilent && problem.marker.severity === DiagnosticSeverity.Error) {
+                            const terminalId = matchedRunningTaskInfo!.terminalId;
+                            if (terminalId) {
+                                const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                                if (terminal) {
+                                    const focus = !!matchedRunningTaskInfo!.config.presentation!.focus;
```

Comment:
Did we already update to typescript version which support `!.`?

Reply:
Yes I should have had used `!.` for a while.